test/ringpop-assert.js

```diff
@@ -52,8 +52,10 @@ function waitForJoins(t, tc, n) {
         t.equals(joins.length, n, 'check number of joins',
             errDetails({journal: _.pluck(list, 'endpoint')}));
 
-        //XXX: a bit inappropriate to get sutIncarnationNumber like this
-        tc.test_state['sutIncarnationNumber'] = safeJSONParse(joins[0].arg3).incarnationNumber;
+        //XXX: a bit inappropriate to set internal state like this
+        var joinRequest = safeJSONParse(joins[0].arg3);
```

Comment:
I think you can use `.body` here as a parsed version of `arg3` already. It hasn't been changed throughout the code, but now that you are changing it you might as well use it.